# Aircraft Risk Assessment for Aircraft Investment

## 1. Business Understanding
### 1.1 Background

Huru Capital Investments Group is a Venture Capital firm with a presence in the construction, real estate and logistics sector in the United States of America. Given its strategic goal to diversify its portfolio, the company is exploring expansion into high growth industries such as the aviation sector. This decision is driven by the expected growth in regional and international air transport. 

The Company intends to purchase and operate airplanes for commercial and private use and hence safety is a critical consideration in selecting the right aircraft.

To support this move, Huru Capital Investments Group is leveraging historical aviation accident data from 1962 to 2023 using a [dataset](https://www.kaggle.com/datasets/khsamaha/aviation-accident-database-synopses) from the National Transportation Safety Board to evaluate potential risks associated with various types of aircraft. This analysis will inform executive decision making on which aircraft models offer the best balance between performance and operational safety.

### 1.2 Objectives

The overall objective of this project is to identify the types of aircraft with the lowest historical risk profile based on accident frequency, severity and related factors. These insights will guide Huru Capital's aviation investment strategy by pinpointing safe and reliable aircraft for acquisition.

The specific objectives are:
* To analyse accident frequency and the severity by the type of aircraft
* To identify trends  and patterns related to aviation accidents and incidents over time and geographical location
* To recommend the top three types of aircraft for investment based on safety performance

### 1.3 Problem Statement

Huru Capital Investments Group is interested in entering the aviation industry without any internal expertise on the potential risks of aircraft. Given the potential financial and reputational risks associated with aviation accidents, the company must base its aircraft selection on data-driven insights. 

This analysis will help answer the following question:

> *Which types of aircraft present the lowest investment risk based on historical safety records?*

### 1.4 Metrics of Success

This project will be considered successful if:

* The analysis identifies aircraft types with historically low accident and fatality rates, helping the company make informed investment decisions.
* At least three clear evidence based recommendations are provided to guide aircraft acquisition
* Key safety patterns (by aircraft type, location, cause or time) are clearly visualised making the risks and opportunities easy to interpret
* The insights directly support Huru Capital's broader goal of expanding into new industries with minimised operational and reputational risk

## 2. Data Understanding

This entails an initial data exploration so as to get an understanding of the shape, structure, contents and quality of the data. 

The data to be utilised in this analysis is drawn from the National Transportation Safety Board Database which contains information from 1962 to 2023 on civil aviation accidents and selected incidents within the United States, its territories and possessions, and in international waters. 

The source of the dataset is https://www.kaggle.com/datasets/khsamaha/aviation-accident-database-synopses

The dataset has two csv files:
* AviationData.csv
* USState_Codes.csv

The aim of this is to:
* identify key variables and their data types
* detect missing or inconsistent values
* gain familiarity with trends, distributions or anomalies

In [2]:
#import the relevant python libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

### 2.1 Initial Data Exploration of Aviation Data

In [5]:
#load the aviation dataset
aviation_df = pd.read_csv('Data/AviationData.csv', encoding='latin-1')

C:\Users\rotic\anaconda3\envs\learn-env\lib\site-packages\IPython\core\interactiveshell.py:3145: DtypeWarning: Columns (6,7,28) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          Location        Country Latitude Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States      NaN       NaN          NaN   
1   BRIDGEPORT, CA  United States      NaN       NaN          NaN   
2    Saltville, VA  United States  36.9222  -81.8781          NaN   
3       EUREKA, CA  United States      NaN       NaN          NaN   
4       Canton, OH  United States      NaN       NaN          NaN   

  Airport.Name  ... Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
0          NaN  ...          Personal         NaN                  2.0   
1          NaN  ...          Personal         NaN                  4.0   
2          NaN  ...          Personal         NaN                  3.0   
3          NaN  ...          Personal         NaN                  2.0   
4          NaN  ...          Personal         NaN                  1.0   

  Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
0                    0.0                  0.0             0.0   
1                    0.0                  0.0             0.0   
2                    NaN                  NaN             NaN   
3                    0.0                  0.0             0.0   
4                    2.0                  NaN             0.0   

  Weather.Condition  Broad.phase.of.flight   Report.Status Publication.Date  
0               UNK                 Cruise  Probable Cause              NaN  
1               UNK                Unknown  Probable Cause       19-09-1996  
2               IMC                 Cruise  Probable Cause       26-02-2007  
3               IMC                 Cruise  Probable Cause       12-09-2000  
4               VMC               Approach  Probable Cause       16-04-1980  

[5 rows x 31 columns]

In [15]:
#preview the first 5 rows
aviation_df.head()

Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          Location        Country Latitude Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States      NaN       NaN          NaN   
1   BRIDGEPORT, CA  United States      NaN       NaN          NaN   
2    Saltville, VA  United States  36.9222  -81.8781          NaN   
3       EUREKA, CA  United States      NaN       NaN          NaN   
4       Canton, OH  United States      NaN       NaN          NaN   

  Airport.Name  ... Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
0          NaN  ...          Personal         NaN                  2.0   
1          NaN  ...          Personal         NaN                  4.0   
2          NaN  ...          Personal         NaN                  3.0   
3          NaN  ...          Personal         NaN                  2.0   
4          NaN  ...          Personal         NaN                  1.0   

  Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
0                    0.0                  0.0             0.0   
1                    0.0                  0.0             0.0   
2                    NaN                  NaN             NaN   
3                    0.0                  0.0             0.0   
4                    2.0                  NaN             0.0   

  Weather.Condition  Broad.phase.of.flight   Report.Status Publication.Date  
0               UNK                 Cruise  Probable Cause              NaN  
1               UNK                Unknown  Probable Cause       19-09-1996  
2               IMC                 Cruise  Probable Cause       26-02-2007  
3               IMC                 Cruise  Probable Cause       12-09-2000  
4               VMC               Approach  Probable Cause       16-04-1980  

[5 rows x 31 columns]

In [16]:
#preview the last 5 rows
aviation_df.tail()

Event.Id Investigation.Type Accident.Number  Event.Date  \
88884  20221227106491           Accident      ERA23LA093  2022-12-26   
88885  20221227106494           Accident      ERA23LA095  2022-12-26   
88886  20221227106497           Accident      WPR23LA075  2022-12-26   
88887  20221227106498           Accident      WPR23LA076  2022-12-26   
88888  20221230106513           Accident      ERA23LA097  2022-12-29   

            Location        Country Latitude Longitude Airport.Code  \
88884  Annapolis, MD  United States      NaN       NaN          NaN   
88885    Hampton, NH  United States      NaN       NaN          NaN   
88886     Payson, AZ  United States  341525N  1112021W          PAN   
88887     Morgan, UT  United States      NaN       NaN          NaN   
88888     Athens, GA  United States      NaN       NaN          NaN   

      Airport.Name  ... Purpose.of.flight         Air.carrier  \
88884          NaN  ...          Personal                 NaN   
88885          NaN  ...               NaN                 NaN   
88886       PAYSON  ...          Personal                 NaN   
88887          NaN  ...          Personal  MC CESSNA 210N LLC   
88888          NaN  ...          Personal                 NaN   

      Total.Fatal.Injuries Total.Serious.Injuries Total.Minor.Injuries  \
88884                  0.0                    1.0                  0.0   
88885                  0.0                    0.0                  0.0   
88886                  0.0                    0.0                  0.0   
88887                  0.0                    0.0                  0.0   
88888                  0.0                    1.0                  0.0   

      Total.Uninjured Weather.Condition  Broad.phase.of.flight Report.Status  \
88884             0.0               NaN                    NaN           NaN   
88885             0.0               NaN                    NaN           NaN   
88886             1.0               VMC                    NaN           NaN   
88887             0.0               NaN                    NaN           NaN   
88888             1.0               NaN                    NaN           NaN   

      Publication.Date  
88884       29-12-2022  
88885              NaN  
88886       27-12-2022  
88887              NaN  
88888       30-12-2022  

[5 rows x 31 columns]

#### Observation

The dataset is uniform from top to bottom

In [9]:
#get the shape of the dataframe (no of rows, no of columns)
aviation_df.shape

(88889, 31)

In [6]:
#get the summary of information on the aviation dataframe
aviation_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 88889 entries, 0 to 88888
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      88889 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Latitude                34382 non-null  object 
 7   Longitude               34373 non-null  object 
 8   Airport.Code            50249 non-null  object 
 9   Airport.Name            52790 non-null  object 
 10  Injury.Severity         87889 non-null  object 
 11  Aircraft.damage         85695 non-null  object 
 12  Aircraft.Category       32287 non-null  object 
 13  Registration.Number     87572 non-null  object 
 14  Make                    88826 non-null

#### Observation

The Aviation Dataframe has **88889 rows** and **31 columns**.
It is composed of two data types:
* Numerical data - Float (5 columns)
* Categorical data - Object (26 columns)

27 columns have null values (float(5) and object(22)) with varying null values. The key columns with null values that will need to be cleaned/explored for analysis are:
* Severity of Injuries
* Total Fatal Injuries
* Total Serious Injuries
* Total Minor Injuries
* Total Uninjured
* Aircraft Make
* Aircraft Model
* Location
* Event Date
* Weather Condition
* Number of Engines
* Type of Engines
* Purpose of Flight
* Broad Phase of Flight

There is information on date provided under the Event Date and Publication Date columns which is represented as object data type but should be converted into date-time data type to allow for trend analysis.

The Location column contains information in both uppercase and lowercase which should be standardised for parsing.

In [17]:
# getting concise statistical summary of numerical data
aviation_df.describe().T

count      mean        std  min  25%  50%  75%  \
Number.of.Engines       82805.0  1.146585   0.446510  0.0  1.0  1.0  1.0   
Total.Fatal.Injuries    77488.0  0.647855   5.485960  0.0  0.0  0.0  0.0   
Total.Serious.Injuries  76379.0  0.279881   1.544084  0.0  0.0  0.0  0.0   
Total.Minor.Injuries    76956.0  0.357061   2.235625  0.0  0.0  0.0  0.0   
Total.Uninjured         82977.0  5.325440  27.913634  0.0  0.0  1.0  2.0   

                          max  
Number.of.Engines         8.0  
Total.Fatal.Injuries    349.0  
Total.Serious.Injuries  161.0  
Total.Minor.Injuries    380.0  
Total.Uninjured         699.0

In [18]:
# getting concise statistical summary of categorical data
aviation_df.describe(include='O').T

count unique             top   freq
Event.Id               88889  87951  20001214X45071      3
Investigation.Type     88889      2        Accident  85015
Accident.Number        88889  88863      WPR22LA201      2
Event.Date             88889  14782      1984-06-30     25
Location               88837  27758   ANCHORAGE, AK    434
Country                88663    219   United States  82248
Latitude               34382  25592         332739N     19
Longitude              34373  27156        0112457W     24
Airport.Code           50249  10375            NONE   1488
Airport.Name           52790  24871         Private    240
Injury.Severity        87889    109       Non-Fatal  67357
Aircraft.damage        85695      4     Substantial  64148
Aircraft.Category      32287     15        Airplane  27617
Registration.Number    87572  79105            NONE    344
Make                   88826   8237          Cessna  22227
Model                  88797  12318             152   2367
Amateur.Built          88787      2              No  80312
Engine.Type            81812     13   Reciprocating  69530
FAR.Description        32023     31             091  18221
Schedule               12582      3            NSCH   4474
Purpose.of.flight      82697     26        Personal  49448
Air.carrier            16648  13590           Pilot    258
Weather.Condition      84397      4             VMC  77303
Broad.phase.of.flight  61724     12         Landing  15428
Report.Status          82508  17075  Probable Cause  61754
Publication.Date       75118   2924      25-09-2020  17019

### 2.2 Initial Data Exploration of US State Codes data

In [10]:
#Load the US State Codes dataset
USStateCodes_df = pd.read_csv('Data/USState_Codes.csv', encoding='latin-1')

US_State Abbreviation
0     Alabama           AL
1      Alaska           AK
2     Arizona           AZ
3    Arkansas           AR
4  California           CA

In [13]:
#Preview the first 5 rows
USStateCodes_df.head()

US_State Abbreviation
0     Alabama           AL
1      Alaska           AK
2     Arizona           AZ
3    Arkansas           AR
4  California           CA

In [14]:
USStateCodes_df.tail()

US_State Abbreviation
57  Virgin Islands           VI
58   Washington_DC           DC
59  Gulf of mexico           GM
60  Atlantic ocean           AO
61   Pacific ocean           PO

#### Observation

The dataset is uniform from top to bottom

In [11]:
USStateCodes_df.shape

(62, 2)

In [12]:
USStateCodes_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 62 entries, 0 to 61
Data columns (total 2 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   US_State      62 non-null     object
 1   Abbreviation  62 non-null     object
dtypes: object(2)
memory usage: 1.1+ KB


#### Observation

The US State Codes Dataframe has **62 rows** and **2 columns**. The data type of the columns is categorical(object). There are no null values.